pkg/tasks/tasks.go

```diff
@@ -292,6 +292,35 @@ func WithReset(t Tasks) Tasks {
 	}...)
 }
 
+func WithContainerDMigration(t Tasks) Tasks {
+	return WithHostnameOS(t).
+		append(Tasks{
+			{Fn: validateContainerdInConfig, ErrMsg: "failed to validate config", Retries: 1},
+			{Fn: kubeconfig.BuildKubernetesClientset, ErrMsg: "failed to build kubernetes clientset"},
+			{Fn: migrateToContainerd, ErrMsg: "failed to migrate to containerd"},
+			{Fn: patchCRISocketAnnotation, ErrMsg: "failed to patch Node objects"},
+			{
+				Fn: func(s *state.State) error {
+					s.Logger.Info("Downloading PKI...")
+					return s.RunTaskOnLeader(certificate.DownloadKubePKI)
+				},
+				ErrMsg: "failed to download Kubernetes PKI from the leader",
+			},
+			{
+				Fn: func(s *state.State) error {
+					if err := machinecontroller.Ensure(s); err != nil {
+						return err
+					}
+
+					s.Logger.Warn("roll-over your machineDeployments to get containerd")
```

Comment:
Tiny nit: we usually start log messages with a capital letter. Also, what do you think about rephrasing this message to include some more details?

```suggestion
					s.Logger.Warn("Roll-over your MachineDeployments to migrate the worker nodes to containerd")
```

Reply:
OK, changed that to a link to doc with how to launch a rolling restart of machineDeployment.